streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -545,13 +548,33 @@ public Subscription subscription(Set<String> topics) {
 
                 // finally, encode the assignment before sending back to coordinator
                 assignment.put(consumer, new Assignment(activePartitions, new AssignmentInfo(active, standby, partitionsByHostState).encode()));
-                i++;
             }
         }
 
         return assignment;
     }
 
+    // visible for testing
+    List<List<TaskId>> interleaveTasksByGroupId(Collection<TaskId> taskIds, int numberThreads) {
+        final List<TaskId> sortedTasks = new ArrayList<>(taskIds);
+        Collections.sort(sortedTasks);
+        List<List<TaskId>> taskIdsForConsumerAssignment = new ArrayList<>(numberThreads);
+        for (int i = 0; i < numberThreads; i++) {
+            taskIdsForConsumerAssignment.add(new ArrayList<TaskId>());
+        }
+        LinkedList<TaskId> taskIdLinkedList = new LinkedList<>(sortedTasks);
+        while (!taskIdLinkedList.isEmpty()) {
+            for (List<TaskId> taskIdList : taskIdsForConsumerAssignment) {
+                TaskId taskId = taskIdLinkedList.poll();
+                if (taskId != null) {
```

Comment:
If `taskId == null` we should call `break` to terminate instead of finish the whole loop.

Reply:
ack